server/athenian/api/align/queries/metrics.py

```diff
@@ -215,14 +217,31 @@ def __eq__(self, other: object) -> bool:
         return self.team_id == other.team_id and self.goal_id == other.goal_id
 
 
-TeamMetricsResult = dict[Intervals, dict[str, dict[tuple[int, int], list[Any]]]]
+@dataclass(frozen=True, slots=True, order=True)
+class MetricWithParams:
+    """A metric with the optional associated parameters."""
+
+    metric: str
+    metric_params: Optional[dict[str, Hashable]] = field(compare=False)
+    metric_params_tuple: Optional[tuple[tuple[str, Hashable], ...]] = field(init=False, repr=False)
```

Comment:
```suggestion
    parameters: dict[str, Hashable]
```
- Metrics with different parameters must behave differently and should participate in the comparisons.
- Empty dicts for empty parameters, no 3-state objects.
- parameter name -> value, the simplest logic and the only way to code this without overcomplicating. We should never require any parameter ordering.
- no exotic `field`-s